### PR TITLE
Add unit test metaschema conformant with ADR#6

### PR DESCRIPTION
**Pull request by aj-stein-gsa**
# Committer Notes

This PR will add a machine-readable specification of the JSON/YAML-based unit-test initiated in #593. It does not include proposed changes with team support in #817. Such a change will come in a later PR.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\] (Will squash/rebase on approval and merge.)
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- ~If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~ This is a backlog item to backport work into gaps from previous and current issues in progress on the board.
- ~If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?~ This is a backlog item to backport work into gaps from previous and current issues in progress on the board.

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by Gabeblis — DISMISSED** [review]
This looks really good! This actually is already compatible with the change I made in #817. This would be useful to integrate into the GitHub Actions workflows.

**Comment by aj-stein-gsa** [comment]
> This looks really good! This actually is already compatible with the change I made in #817. This would be useful to integrate into the GitHub Actions workflows.

Yeah turns out making it less compatible was annoying, and I found a resolved bug in #219 and a sorta bug (really a "well, don't do that" that will lead to fixups in #221, but yep, there you have it.

**Review by Rene2mt — CHANGES_REQUESTED** [review]
@aj-stein-gsa this is great!!!  Liking the application of metaschema and metaschema constraints  this scenario. 

I have a blocking change request to include the `pipeline` assembly reference, but aside from that, I tested locally  and it looks good.

File: src/validations/constraints/unit-tests/unit_test_metaschema.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    <define-assembly name="test-case">
+        <formal-name>Test Case Document</formal-name>
+        <description>This is the root element of the FedRAMP Automation Test Suite format for collections of unit tests.</description>
+        <root-name>test-case</root-name>
+        <model>
+            <field ref="name" min-occurs="1" max-occurs="1"/>
+            <field ref="description" min-occurs="1" max-occurs="1"/>
+            <field ref="content" min-occurs="0" max-occurs="unbounded">
+                <group-as name="content" in-json="SINGLETON_OR_ARRAY"/>
+            </field>
```
[the remark this answers is not in the document]
Comment (aj-stein-gsa) [reply src/validations/constraints/unit-tests/unit_test_metaschema.xml:19]: Excellent catch! Sorry I had to comment some things out for debugging and testing upstream yesterday and I didn't put it back.

Gold star for @Rene2mt!

**Review by Rene2mt — APPROVED** [review]

**Review by DimitriZhurkin — APPROVED** [review]

**Review by wandmagic — COMMENTED** [review]
nice work, I'll add this to the test runner

**Pull request merged**